assertion_test.go

```diff
@@ -195,7 +195,205 @@ func TestAssertion_HandlerPanics(t *testing.T) {
 	})
 }
 
-func TestAssertion_Validation(t *testing.T) {
+func TestAssertion_ValidateTraits(t *testing.T) {
+	tests := []struct {
+		testName          string
+		errorContainsText string
+		failure           AssertionFailure
+		traits            fieldTraits
+	}{
+		{
+			testName:          "bad Type",
+			errorContainsText: "AssertionType",
+			failure: AssertionFailure{
+				Type: AssertionType(9999),
+			},
+			traits: fieldTraits{
+				Actual: fieldRequired,
+			},
+		},
+		{
+			testName:          "required Actual",
+			errorContainsText: "Actual",
+			failure: AssertionFailure{
+				Actual: nil,
+			},
+			traits: fieldTraits{
+				Actual: fieldRequired,
+			},
+		},
+		{
+			testName:          "denied Actual",
+			errorContainsText: "Actual",
+			failure: AssertionFailure{
+				Actual: &AssertionValue{},
+			},
+			traits: fieldTraits{
+				Actual: fieldDenied,
+			},
+		},
+		{
+			testName:          "required Expected",
+			errorContainsText: "Expected",
+			failure: AssertionFailure{
+				Expected: nil,
+			},
+			traits: fieldTraits{
+				Expected: fieldRequired,
+			},
+		},
+		{
+			testName:          "denied Expected",
+			errorContainsText: "Expected",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{},
+			},
+			traits: fieldTraits{
+				Expected: fieldDenied,
+			},
+		},
+		{
+			testName:          "required on Expected",
+			errorContainsText: "",
+			failure: AssertionFailure{
+				Expected: nil,
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "AssertionRange should not be pointer",
+			errorContainsText: "AssertionRange",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: &AssertionRange{},
+				},
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "Min should be non-nil",
+			errorContainsText: "AssertionRange",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionRange{
+						Max: 1,
+					},
+				},
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "Max should be non-nil",
+			errorContainsText: "AssertionRange",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: AssertionRange{
+						Min: 1,
+					},
+				},
+			},
+			traits: fieldTraits{
+				Range: fieldRequired,
+			},
+		},
+		{
+			testName:          "AssertionList should not be pointer",
+			errorContainsText: "AssertionValue",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: &AssertionList{},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
+		{
+			testName:          "AssertionList should be pointer",
+			errorContainsText: "AssertionList",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{
+					Value: &AssertionList{},
+				},
+			},
+			traits: fieldTraits{
+				List: fieldRequired,
+			},
+		},
```

Comment:
Seems this test duplicates previous one and can be removed. Also its name is misleading.